### PR TITLE
Record: Int6 + MLP 3x + NorMuon + SmearGate + BigramHash + OrthoInit + Sliding Window, val_bpb=1.1541

### PR DESCRIPTION
**Pull request by MatthewHRockwell**
## Summary

**val_bpb: 1.1541** (sliding window stride=256) | **Artifact: 15,992,610 bytes**

Seven improvements stacked on the baseline:

1. **Int6 per-row quantization** — [-31,31] with fp16 scales, fp16 tied embedding. Quant gap: 0.011 BPB
2. **MLP 3x expansion** — hidden=1536, enabled by int6 space savings
3. **NorMuon + decoupled weight decay** (0.02) — per-row normalized Newton-Schulz
4. **SmearGate** — learned gate blending with previous token (~512 params)
5. **BigramHash** — 4096-bucket hash embedding for token pairs (dim=64)
6. **Orthogonal + muP init** — orthogonal init + 1/sqrt(2L) scaling on output projections
7. **Sliding window eval** — stride=256, 2048-token windows, distributed

### Key Metrics

| Metric | Value |
|--------|-------|
| Pre-quant val_bpb | 1.1644 |
| Int6 roundtrip val_bpb | 1.1752 |
| **Sliding window val_bpb** | **1.1541** |
| Steps | 7,332 @ 81.84ms/step |
| Tokens | ~5.77B |
| Artifact | 15,992,610 bytes |

Training: `TRAIN_SEQ_LEN=2048 TRAIN_BATCH_TOKENS=786432 MATRIX_LR=0.02 MUON_MOMENTUM=0.99 WARMDOWN_ITERS=3000 GRAD_CLIP_NORM=1.0`

```bash
NCCL_IB_DISABLE=1 torchrun --standalone --nproc_per_node=8 train_gpt.py
```

8xH100 SXM (RunPod). Full log included.

## Test plan

- [x] Full pipeline: train → eval → int6 export → roundtrip → sliding window eval
- [x] Artifact under 16MB (15.99MB)
- [x] 8xH100 SXM verified
- [x] Runs within 10-min wallclock + 10-min eval budget

### CODE REVIEW

**Review by MatthewHRockwell — COMMENTED** [review]
Tested

**Comment by MatoTeziTanka** [comment]
### Community Review — Record: Int6 + MLP 3x + NorMuon + SmearGate + BigramHash + OrthoInit + Sliding Window, val_bpb=1.1541

**BPB:** 1.1541 | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `f0ab434a41a6`, file `records/track_10min_16mb/2026-03-20_Int6_MLP3x_NorMuon_WD/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.10s, dim=512, layers=9, vocab=1024, code=53199 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.10s, dim=512, layers=9, vocab=1024, code=53199 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
